### PR TITLE
Add support for reproducible builds

### PR DESCRIPTION
**Pull request by inglor**
### Description
Add support for reproducible builds

As per gradle [docs] add support to remove timestamps and package with same order which is required from
[reproducible] builds

[docs]: https://docs.gradle.org/current/userguide/working_with_files.html#sec:archives
[reproducible]: https://reproducible-builds.org/

### Issues Resolved
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by amitgalitz** [comment]
Hi @inglor, thanks for providing links for documentation on reproducible build and how they are useful. I was wondering if you have any further particular reasoning for adding it to Anomaly Detection at the time, do you have a use case for this? Just to understand more context for this change.

**Comment by inglor** [comment]
Hi @amitgalitz, see https://github.com/opensearch-project/OpenSearch/pull/1995#issuecomment-1023744621. Similarly this project fails unless the above change is applied.
As part of my efforts to make OpenSearch reproducible in Arch Linux I am making PRs to each of the subprojects (slowly). Still got some to do [1] 

[1]: https://archlinux.org/devel/reports/non-reproducible-packages/artafinde/

**Comment by amitgalitz** [comment]
Just to confirm since we don't utilize the Gradle GlobalBuildInfo plugin, we don't need the other change you introduced in the OS PR?

**Comment by inglor** [comment]
@amitgalitz Correct. This change is tested and produces reproducible builds as-is

**Review by amitgalitz — APPROVED** [review]
LGTM, thanks for adding this! Will need one more approval before merging

**Review by ohltyler — APPROVED** [review]

**Comment by dblock** [comment]
@inglor Have you looked into making a change in OpenSearch `opensearch.plugin` task(s) so that the plugins can automatically inherit this and avoid making the PRs in every plugin?

**Comment by dblock** [comment]
I raised https://github.com/opensearch-project/opensearch-plugin-template-java/pull/38#issuecomment-1262592602, maybe we shouldn't put this in every plugin repo.

**Comment by prudhvigodithi** [comment]
Added some thoughts https://github.com/opensearch-project/opensearch-plugin-template-java/pull/38#issuecomment-1264690748, may be we should consider putting this reproducible build logic to one of the common custom gradle plugins used.

**Pull request merged**
